clients/js/src/cipher.ts

```diff
@@ -251,11 +258,14 @@ export class X25519DeoxysII extends Cipher {
     this.cipher = new deoxysii.AEAD(new Uint8Array(this.key)); // deoxysii owns the input
   }
 
-  public encrypt(plaintext: Uint8Array): {
+  public encrypt(
+    plaintext: Uint8Array,
+    nonce?: Uint8Array,
```

Comment:
```suggestion
    nonce: Uint8Array = randomBytes(deoxysii.NonceSize),
```
Instead of check and assign, it could be defaulted to randomBytes directly